Source/Scene/ShadowMapShader.js

```diff
@@ -38,41 +38,120 @@ define([
         return fs;
     };
 
-    ShadowMapShader.createReceiveShadowsVertexShader = function(vs) {
+    ShadowMapShader.createShadowReceiveVertexShader = function(vs) {
         vs = ShaderSource.replaceMain(vs, 'czm_shadow_main');
         vs +=
-            'varying vec3 czm_shadowMapCoordinate; \n' +
+            'varying vec3 v_shadowPosition; \n' +
             'void main() \n' +
             '{ \n' +
             '    czm_shadow_main(); \n' +
-            '    czm_shadowMapCoordinate = (czm_sunShadowMapMatrix * gl_Position).xyz; \n' +
+            '    v_shadowPosition = (czm_sunShadowMapMatrix * gl_Position).xyz; \n' +
             '} \n';
         return vs;
     };
 
-    ShadowMapShader.createReceiveShadowsFragmentShader = function(fs, context) {
+    ShadowMapShader.createShadowReceiveFragmentShader = function(fs, context) {
+        var cascadesEnabled = true;
+        var debugVisualizeCascades = true;
         fs = ShaderSource.replaceMain(fs, 'czm_shadow_main');
         fs +=
-            'varying vec3 czm_shadowMapCoordinate; \n' +
+            'varying vec3 v_shadowPosition; \n' +
+            ' \n' +
+            'vec4 getCascadeWeights(float depthEye) \n' +
+            '{ \n' +
+            '    // One component is set to 1.0 and all others set to 0.0. \n' +
+            '    vec4 near = step(czm_sunShadowMapCascadeSplitsNear, vec4(depthEye)); \n' +
+            '    vec4 far = step(depthEye, czm_sunShadowMapCascadeSplitsFar); \n' +
+            '    return near * far; \n' +
+            '} \n' +
+            'vec4 getCascadeViewport(vec4 weights) \n' +
+            '{ \n' +
+            '    return vec4(0.0, 0.0, 0.5, 0.5) * weights.x + \n' +
+            '           vec4(0.5, 0.0, 0.5, 0.5) * weights.y + \n' +
+            '           vec4(0.0, 0.5, 0.5, 0.5) * weights.z + \n' +
+            '           vec4(0.5, 0.5, 0.5, 0.5) * weights.w; \n' +
+            '} \n' +
+            'vec3 getCascadeOffset(vec4 weights) \n' +
+            '{ \n' +
+            '    return czm_sunShadowMapCascadeOffsets[0] * weights.x + \n' +
```

Comment:
`czm_sunShadowMapCascadeOffsets` and `czm_sunShadowMapCascadeScales` are arrays of vec3 so I don't think that would work.